clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -501,6 +506,31 @@ void CIRGenModule::setCommonAttributes(GlobalDecl GD, mlir::Operation *GV) {
   assert(!UnimplementedFeature::setCommonAttributes());
 }
 
+void CIRGenModule::replaceGlobal(mlir::cir::GlobalOp Old,
+                                 mlir::cir::GlobalOp New) {
+  assert(Old.getSymName() == New.getSymName() && "symbol names must match");
+
+  // If the types does not match, update all references to Old to the new type.
+  auto OldTy = Old.getSymType();
+  auto NewTy = New.getSymType();
+  if (OldTy != NewTy) {
+    auto OldSymUses = Old.getSymbolUses(theModule.getOperation());
+    if (OldSymUses.has_value()) {
+      for (auto Use : *OldSymUses) {
+        auto UseOp = dyn_cast<mlir::cir::GetGlobalOp>(Use.getUser());
+        assert(UseOp && "GlobalOp symbol user is not a GetGlobalOp");
```

Comment:
Seems this `assert` is too restrictive. Besides `GlobalOp`, we may also have `GlobalViewAttr` which can be used in `cir.global`:

```cpp
extern int table[];

int *table_ptr = table;
// cir.global external @table_ptr = #cir.global_view<@table> : !cir.ptr<!s32i>

int table[3] {1, 2, 3};
```

Reply:
Updated.

Reply:
Updated.